spec/unit/knife/bootstrap_spec.rb

```diff
@@ -593,6 +593,20 @@ def certificates
         knife.run
       end
     end
+
+    context "when the validation_key is nil" do
+      before do
+        # this tests runs the old code path where we have a validation key, so we need to pass that check for some plugins
+        Chef::Config[:validation_key] = nil
+      end
+
+      it "creates the client (and possibly adds chef-vault items)" do
+        expect(knife_ssh).to receive(:run)
+        expect(knife.client_builder).not_to receive(:run)
+        expect(knife.chef_vault_handler).not_to receive(:run).with(node_name: knife.config[:chef_node_name])
```

Comment:
Is receiving `:run` ok if node_name is something other than `knife.config[:chef_node_name]`